packages/ibm-products/src/global/js/hooks/useOverflowItems/useOverflowItems.mdx

```diff
@@ -0,0 +1,73 @@
+import { Story, Canvas } from '@storybook/addon-docs';
+import * as UseOverflowItemsStories from './useOverflowItems.stories';
+import { useOverflowItems } from '.';
+
+# useOverflowItems
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Hook API](#hook-api)
+- [Example usage](#example-usage)
+- [Example implementation](#example-implementation)
+
+## Overview
+
+The `useOverflowItems` hook can be used for handling overflow behavior of multiple DOM elements inside of a given area.
+
+For example, if you need to display a certain number of `Tag` elements in a container and render additional values in a popover, you could use this hook to do just that.
+
+## Hook API
+
+```jsx
+// Hook arguments
+items: T[] = [], // An iterable list of items
+containerRef: RefObject<HTMLElement | null>, // The parent container that the items are confined to
+offsetRef?: RefObject<HTMLElement | null>, // An optional ref to an element that you want to exclude from the calculation
+maxItems?: number, // An optional argument for a maximum amount of items to return
+onChange?: (value: {
+  hiddenItems?: T[];
+  minWidth?: number;
+  maxWidth?: number;
+}) => void // An event handler that's called whenever the number of number of visible items changes
+
+// Hook returns
+{
+  visibleItems: T[]; // An array of items that fit in the containerRef
+  itemRefHandler: (id: string, node: HTMLElement | null) => void; // A handler for passing the ref of the node to the hook
+  hiddenItems: T[]; // An array of items that don't fit in the containerRef
+  offsetRefHandler: (node: HTMLElement | null) => HTMLElement; // A handler for passing the ref of the offset element to the hook
+}
+```
+
+## Example usage
+
+```jsx
+const {
+  visibleItems,
+  hiddenItems,
+  itemRefHandler,
+  offsetRefHandler
+} = useOverflowItems(
+  items,
+  containerRef,
+  offsetRef,
+  maxVisible,
+  onChange
+);
```

Comment:
I'm wondering if the example usage should also include a quick tag example since you mentioned it? would be easier to understand the component that way. WDYT?

Reply:
can you clarify? because directly underneath it is the story example
<img width="1840" alt="Screenshot 2025-04-04 at 1 47 50 PM" src="https://github.com/user-attachments/assets/63c67e5e-04f5-4646-90a1-69e2cfffef3c" />